app/services/applications/change_funded_place.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Applications
+  class ChangeFundedPlace
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :application
+    attribute :funded_place, :boolean
+
+    validates :application, presence: { message: I18n.t("application.missing_application") }
+    validates :funded_place,
+              inclusion: {
+                in: [true, false],
+                message: I18n.t("application.missing_funded_place"),
+              }
+    validate :accepted_application
+    validate :eligible_for_funding
+    validate :cohort_has_funding_cap
+
+    def change
+      return false unless valid?
+
+      application.update!(funded_place:)
+    end
+
+  private
+
+    delegate :cohort, to: :application
+
+    def accepted_application
+      return if application.accepted?
+
+      errors.add(:application, I18n.t("application.cannot_change_funded_status_from_non_accepted"))
+    end
+
+    def eligible_for_funding
+      return unless funded_place
+      return if application.eligible_for_funding?
+
+      errors.add(:application, I18n.t("application.cannot_change_funded_status_non_eligible"))
+    end
+
+    def cohort_has_funding_cap
+      return if errors.any?
```

Comment:
we would like to avoid chaining too many errors for providers, it gets confusing if we surface all errors although the main attribute it missing in a request

Reply:
Ah that makes sense. One for another day, but it doesn't make sense to do it here IMO; this is likely to get accidentally removed at some point. I think it would be better implemented in `API::Errors::Response` if its a general rule.

Reply:
I think its not general sometimes we want to chain as it makes sense, and sometimes we don't. However happy to revisit if there is a nicer way of doing this after we add more actions